inductiva/resources/disk_config.py

```diff
@@ -0,0 +1,48 @@
+"""
+This module contains the DiskConfig class which is used to store the
+configuration for the disk.
+"""
+
+from dataclasses import dataclass, field
+
+
+@dataclass(frozen=True)
+class DiskConfig:
+    """Dataclass to represent the disk configuration.
+    
+    A disk has a size in GB and can be resized based on the free space left.
+    To enable resizing, the following attributes must be provided:
+    resize_trigger_gb, resize_increment_gb, and max_size_gb.
+
+    The disk will then start with size_gb and will resize when the free space
+    falls below resize_trigger_gb. The disk will grow by resize_increment_gb
+    during each resize, but will not grow beyond max_size_gb.
+
+    Attributes:
+        size_gb (int): Disk size in GB.
+        max_size_gb (int): Maximum allowed disk size in GB. Disk will not grow
+            beyond this limit.
+    """
+    max_size_gb: int
+
+    # resize_trigger_gb (int): Free space threshold (in GB) to trigger
+    #     resizing. Example: If set to 5GB, the disk will resize when 5GB of
+    #     free space remains.
+    resize_trigger_gb: int = field(default=5, init=False)
+
+    # resize_increment_gb (int): Amount (in GB) to increase the disk size
+    #     during each resize.
+    resize_increment_gb: int = field(default=10, init=False)
+    is_resizable: bool = True
+
+    @property
+    def resize_config(self):
+        """Return the disk resize configuration as a dictionary.
+        If the disk is not resizable, it will return None."""
```

Comment:
why return None? It simply return a dict representation of the object.

Reply:
I did this to avoid logic when creating the request to send to the back end. I will change the method to be called resizing_disk_config, if the disk is not resizable it makes sense that resizing_disk_config is None.
Like this i just need to have 
```
instance_group_config = inductiva.client.models.VMGroupConfig(
            ...
            dynamic_disk_resize_config=self.disk_config.resizing_disk_config,
            **kwargs,
        )
```
Like this there is no need for if's here